File: LeetCodeRecord/MethodFirst/26-RemoveDuplicates.java
package MethodFirst;

class RemoveDuplicates {
    public int removeDuplicates(int[] nums) {
        int i = 0;
        for (int j = 0; j < nums.length; j++) {
            if (nums[i] != nums[j]) {
                if (j - i > 1) {
                    nums[i + 1] = nums[j];
                }

                i++;
            }
        }

        return i + 1;
    }    
}

File: LeetCodeRecord/ACFirst-Golang/26-RemoveDuplicates.go
package main

func removeDuplicates(nums []int) int {
	i := 0
	j := 1

	for j < len(nums) {
		if nums[i] != nums[j] {
			nums[i+1] = nums[j]
			i++
			j++
		} else {
			j++
		}
	}

	return i + 1
}
